### ISSUE TITLE
Display all typescript errors and warning of a folder/project

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode-tslint#54]
_From @llgcode on June 8, 2016 7:17_

- VSCode Version: 1.2
- OS Version: Windows 7

It would be useful to have a separate command to display all errors and warnings (tslint) of a typescript project. Can't see how to do this in an integrated way.

_Copied from original issue: Microsoft/vscode#7388_

**Comment by dbaeumer** [comment]
@llgcode you can setup up a task that runs tslint over the entire workspace using gulp, jake, or evem npm. 

Will move the issue to tslint since project validation support for tslint needs to be implemented there. For the TS compiler we do have a similar request.

**Comment by egamma** [comment]
@llgcode this is how we run tslint for the vscode project:
the tasks.json: https://github.com/Microsoft/vscode/blob/master/.vscode/tasks.json#L39
the glup task: https://github.com/Microsoft/vscode/blob/master/build/gulpfile.hygiene.js#L111

**Comment by llgcode** [comment]
Ok so I need to use gulp to have multiple task, running ts and tslint.

**Comment by egamma** [comment]
@llgcode 
>Is it mandatory to have gulp for this?

Now you can use other task runners, e.g. grunt.

**Comment by meriturva** [comment]
@egamma Ok i have configured gulp to run tslint on all project and i have configured same reporter has Microsoft project. But how to integrate output with "error" window? i mean... it could be really useful to have problems listed on that view and access to file with a simple dbClick
What do you think?

**Comment by llgcode** [comment]
Hi @meriturva,
I don't use gulp but I've configure my task runner with a custom problemMatcher like this so it can be accessed by clicking in the error/warning output of vscode 
```js
{
  "version": "0.1.0",
  "command": "cmd",
  "isShellCommand": true,
  "args": ["/C"],
  "showOutput": "silent",
  "tasks": [
...
    {
        "taskName": "tslint",
        "suppressTaskName": true,
        "args": ["tslint -c ../../git/MyConfig/js/tslint.json  /**/*.ts"],
        "problemMatcher": {
           "owner": "tslint",
           "fileLocation": [
               "relative",
               "${workspaceRoot}"
           ],
           "severity":  "warning",
           "pattern": {
               "regexp": "^(.*)\\[(\\d+), (\\d+)\\]:\\s+(.*)$",
               "file": 1,
               "line": 2,
               "column": 3,
               "message": 4
           }
        }
    }
...
]
}
```

**Comment by meriturva** [comment]
Hi @llgcode,
so thanks! it works....
my fault was leaving custom reporter on tslint gulp task. With default reporter everything is ok.
So anyone using gulp....just use default reporter and leave pattern on task.json file as @llgcode wrote!
Again thanks!

**Comment by egamma** [comment]
@meriturva can you please review the README in the latest tslint version. I've added a section for how to set this up.

**Comment by meriturva** [comment]
@egamma Using method described we had two main problem. First: what i have noted when you have a lot of errors is that using reporter provided make all process so slow...Second: pattern on vscode task.json dosen't get error on vscode console. So what i prefer is to leave gulp task simple...what do you think?

**Comment by egamma** [comment]
@meriturva I prefer simple as well. Can you share the gulp task and task definition that works better for you?

**Comment by meriturva** [comment]
Here gulp:
```
gulp.task('VisualStudio:lint-ts', 'Run TsLint and generate output for Visual Studio Console', function (callback) {
    var tslintConfig = require('./tslint.json');
    return gulp.src(['**/*.ts', '!**/*.d.ts', '!node_modules/**','!dist/**'])
      .pipe(gulpTslint({
           configuration: tslintConfig
      }))
      .pipe(gulpTslint.report());
});
```

Here task.json:
```
"taskName": "VisualStudio:lint-ts",
            "args": [],
            "problemMatcher": {
				"owner": "tslint",
				"fileLocation": [
					"relative",
					"${workspaceRoot}"
				],
				"severity":  "warning",
           "pattern": {
               "regexp": "^(.*)\\[(\\d+), (\\d+)\\]:\\s+(.*)$",
               "file": 1,
               "line": 2,
               "column": 3,
               "message": 4
           }
			}
```

**Comment by egamma** [comment]
@meriturva thanks, this is simpler I'll update the docs.

**Comment by egamma** [comment]
Updated the README with a simpler version based on the setup from @meriturva.
